pandas/tests/test_reshape.py

```diff
@@ -171,6 +172,26 @@ def test_basic(self):
         expected.index = list('ABC')
         assert_frame_equal(get_dummies(s_series_index, sparse=self.sparse), expected)
 
+    def test_basic_types(self):
+        # GH 10531
+        s_list = list('abc')
+        s_series = Series(s_list)
+        s_df = DataFrame({'a': [0, 1, 0, 1, 2],
+                          'b': ['A', 'A', 'B', 'C', 'C'],
+                          'c': [2, 3, 3, 3, 2]})
+
+        exp_type = DataFrame if not self.sparse else SparseDataFrame
+        self.assertEqual(type(get_dummies(s_list, sparse=self.sparse)), exp_type)
```

Comment:
@jreback Numerous exact expectation tests are already done in `test_basic` and the other methods of `TestGetDummies` (which are run for both `sparse=False` and `sparse=True`).  However, `assert_frame_equal` -- used in those tests -- does not discriminate between `DataFrame` and `SparseDataFrame` (which is why this bug arose in the first place).  So I put in one extra test that just checks the types.